File: important-questions/js/1.LCSOf1s.js
const x = [1, 1, 0, 1, 1, 0, 1];

var y = [];

x.forEach((a, b) => {
  if (b != 0 || b != x.length - 1)
    if (a === 0) {
      y.push(b);
    }
});

y.unshift(0);
y.push(x.length - 1);
let current = 0;
let max = 0;
let diff = 0;

for (let i = 0; i < y.length - 2; i++) {
  diff = y[i + 2] - y[i];
  if (diff > max) {
    max = diff;
    current = y[i + 1];
  }
}

if (y.length == 2 && x[x.length - 1] == 0) current = x.length - 1;

console.log(current);

File: important-questions/kotlin/1.LCSOf1s.kt
fun main(){
    var x= listOf(0,1,1,1,1,1,0,0,1);
    var zeroes= mutableListOf<Int>();
    zeroes.add(0);
    for(i in 1 until  x.size-1){
         if(x[i]==0){
             zeroes.add(i);
         }
    }
    zeroes.add(x.size-1)
    var max=0;
    var diff=0;
    var current =0;

    for(i in 0 until zeroes.size-2){
        diff = zeroes[i + 2] - zeroes[i];
        if (diff > max) {
            max = diff;
            current = zeroes[i + 1];
        }
    }

    if (zeroes.size == 2 && x[x.size - 1] == 0) {
        current = x.size - 1;
    }

    println(current)

}